GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -758,27 +758,46 @@ private void RegisterPlugins()
                 var pluginEntries = PluginRegistry.Plugins
                     .OrderBy(entry => entry.Name, StringComparer.CurrentCultureIgnoreCase);
 
+                // the menu already contains 2 items:
+                //    [0] Plugin Settings
+                //    [1] separator
+                // insert plugins above the separator (the list is in reverse order)
+                int insertOffset = pluginsToolStripMenuItem.DropDownItems.Count - 2;
+
                 foreach (var plugin in pluginEntries)
                 {
-                    // Add the plugin to the Plugins menu, if not already added
-                    if (!existingPluginMenus.Contains(plugin))
+                    // don't add the plugin to the Plugins menu, if already added
+                    if (existingPluginMenus.Contains(plugin))
                     {
-                        var item = new ToolStripMenuItem
-                        {
-                            Text = plugin.Description,
-                            Image = plugin.Icon,
-                            Tag = plugin
-                        };
-                        item.Click += delegate
+                        continue;
+                    }
+
+                    var item = new ToolStripMenuItem
+                    {
+                        Text = plugin.Description,
+                        Image = plugin.Icon,
+                        Tag = plugin
+                    };
+                    item.Click += delegate
+                    {
+                        if (plugin.Execute(new GitUIEventArgs(this, UICommands)))
                         {
-                            if (plugin.Execute(new GitUIEventArgs(this, UICommands)))
-                            {
-                                RefreshRevisions();
-                            }
-                        };
+                            RefreshRevisions();
+                        }
+                    };
+
+                    if (plugin.Name == "Plugin Manager")
```

Comment:
DRY, especially no magic constants.
```suggestion
                    bool isPluginManager = plugin.Name == "Plugin Manager";
                    if (isPluginManager )
```